File: JS/ValidadorDeSenhasComRequisitos.js
/**
 * @version JAVASCRIPT ECMAScript 6
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */

const VALIDO = "Senha valida.";
const INVALIDO = "Senha invalida.";

while ( (pass = gets()) != '' ) console.log( passwordValidator( pass ) ); 

/**
 * Check if password is valid.
 */
function passwordValidator(pwd = "") {
    let result = VALIDO;

    //valida o tamanho
    if (pwd.length < 6 || pwd.length > 32) {
        result = INVALIDO;
    //valida se possui espacos ou caracteres nao alphanumericos
    } else if (/\s/g.test(pwd) || /\W|_/.test(pwd)) {
        result = INVALIDO;
    //valida se possui elementos obrigatorios (maiusculas, minusculas, e numeros)
    } else if (!/[A-Z]/.test(pwd) || !/[a-z]/.test(pwd) || !/[0-9]/.test(pwd)) {
        result = INVALIDO;
    }

    return result;
}
module.exports = {passwordValidator}

File: GO/ValidadorDeSenhasComRequisitos.go
/**
 * @version GO 1.19
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import (
	"regexp"
	"strings"
)

const VALIDO = "Senha valida."
const INVALIDO = "Senha invalida."

/**
 * Check if password is valid.
 */
func passwordValidator(pwd string) string {
	result := VALIDO

	//regex de validacao
	check_invalids, _ := regexp.Compile(`[[:punct:]]`)
	check_low_az, _ := regexp.Compile(`[[:lower:]]`)
	check_up_az, _ := regexp.Compile(`[[:upper:]]`)
	check_num, _ := regexp.Compile(`[[:digit:]]`)

	//valida o tamanho
	if len(pwd) < 6 || len(pwd) > 32 {
		result = INVALIDO
		//valida se possui espacos ou caracteres nao alphanumericos
	} else if check_invalids.MatchString(pwd) || strings.Contains(pwd, " ") {
		result = INVALIDO
		//valida se possui elementos obrigatorios (maiusculas, minusculas, e numeros)
	} else if !check_low_az.MatchString(pwd) || !check_up_az.MatchString(pwd) || !check_num.MatchString(pwd) {
		result = INVALIDO
	}

	return result
}
